logspout/logspout.go

```diff
@@ -224,14 +224,23 @@ func main() {
 		debug("etcd:", connectionString[0])
 		etcd := etcd.NewClient(connectionString)
 		etcd.SetDialTimeout(3 * time.Second)
-		hostResp, err := etcd.Get("/deis/logs/host", false, false)
-		assert(err, "url")
-		portResp, err := etcd.Get("/deis/logs/port", false, false)
-		assert(err, "url")
-		protocol := getEtcdValueOrDefault(etcd, "/deis/logs/protocol", "udp")
-		host := fmt.Sprintf("%s:%s", hostResp.Node.Value, portResp.Node.Value)
-		log.Printf("routing all to %s://%s", protocol, host)
-		router.Add(&Route{Target: Target{Type: "syslog", Addr: host, Protocol: protocol}})
+		go func() {
+			for {
+				hostResp, err := etcd.Get("/deis/logs/host", false, false)
+				assert(err, "url")
+				portResp, err := etcd.Get("/deis/logs/port", false, false)
+				assert(err, "url")
+				protocol := getEtcdValueOrDefault(etcd, "/deis/logs/protocol", "udp")
+				host := fmt.Sprintf("%s:%s", hostResp.Node.Value, portResp.Node.Value)
+				log.Printf("routing all to %s://%s", protocol, host)
+				// NOTE(bacongobbler): router.Add() will overwrite an existing route. No connection
+				// cleanup is required; logspout spawns a one-off connection for every route then
+				// immediately closes them.
+				router.Add(&Route{ID: "etcd", Target: Target{Type: "syslog", Addr: host, Protocol: protocol}})
```

Comment:
This is the most significant change that makes this all possible: by using a static `ID` for the route, we can overwrite it's value since `RouteManager` just stores them in a `map[string][Route]`, where the string is the route's ID.